### DISCUSSION TITLE
Building in Android Studio

### DISCUSSION THREAD

**Discussion by eapdso** [ungoogled-software/ungoogled-chromium-android#62] [Q&A] [answered]
Hi, i'm working on building this project. 
Is there any documentation that can help me import and build this project in Android Studio directly?
I've been struggling to get this imported/built but haven't had any luck and haven't been able to find any methods to do so online. 
Any advice is really appreciated.

**Comment by wchen342** [comment, marked as answer]
It is only possible to import the Java part of the codes by generating a Gradle file. It is not possible to build directly from Android Studio and the debugging of C++ codes is experimental. See [https://chromium.googlesource.com/chromium/src/+/master/docs/android_studio.md#How-It-Works](https://chromium.googlesource.com/chromium/src/+/master/docs/android_studio.md#How-It-Works).

If you want to generate the gradle file, see `misc/android_studio_setup.sh`.
